src/client/sky.cpp

```diff
@@ -833,7 +833,10 @@ void Sky::setStarCount(u16 star_count, bool force_update)
 	// Allow force updating star count at game init.
 	if (m_star_params.count != star_count || force_update) {
 		m_star_params.count = star_count;
-		m_seed = (u64)myrand() << 32 | myrand();
+		if (!seed_set) {
+			m_seed = (u64)myrand() << 32 | myrand();
```

Comment:
Instead of doing this check you can just move the assignment to the constructor.